src/test/schema.test.ts

```diff
@@ -1176,6 +1176,58 @@ suite('JSON Schema', () => {
 		assert.deepEqual([], accesses); // b is not depended anymore
 	});
 
+	test('resetSchema clears current document schema cache when not using $schema property', async function () {
+		const schemaUri = "http://foo/main.schema.json";
+
+		const schemas: { [uri: string]: JSONSchema } = {
+			[schemaUri]: {
+				type: 'object',
+				properties: {
+					bar: {
+						type: 'string'
+					}
+				}
+			}
+		};
+
+		const accesses: string[] = [];
+		const schemaRequestService = newMockRequestService(schemas, accesses);
+		const testDoc = toDocument(JSON.stringify({ bar: 1 }));
+
+		const ls = getLanguageService({ workspaceContext, schemaRequestService });
+
+		// configure the language service to use the schema for the test document
+		ls.configure({
+			schemas: [{
+				uri: schemaUri,
+				fileMatch: [testDoc.textDoc.uri.toString()],
+			}]
+		});
+
+		// check using the existing schema
+		let validation = await ls.doValidation(testDoc.textDoc, testDoc.jsonDoc);
+		assert.deepStrictEqual(validation.map(v => v.message), ['Incorrect type. Expected "string".']);
+		assert.deepStrictEqual([schemaUri], accesses);
+
+		accesses.length = 0;
+
+		// change a schema property and reset the schema
+		schemas[schemaUri] = {
+			type: 'object',
+			properties: {
+				a: {
+					type: 'number'
+				}
+			}
+		};
+		ls.resetSchema(schemaUri);
+
+		// now ensure validation occurs with the new schema
+		validation = await ls.doValidation(testDoc.textDoc, testDoc.jsonDoc);
+		assert.deepStrictEqual(validation.map(v => v.message), []);
```

Comment:
Previously this would use the resource on `this.cachedSchemaForResource` instead of the latest cache.